snuba/datasets/processors/errors_processor.py

```diff
@@ -327,6 +343,23 @@ def extract_stacktraces(
                     frame_linenos.append(_collapse_uint32(frame.get("lineno", None)))
                     frame_stack_levels.append(stack_level)
 
+                ## mark if one of the exceptions is in the main thread
+                if exception_main_thread is None and thread_id is not None:
+                    for thread in threads:
+                        if thread is None:
+                            continue
+
+                        ## if it's not the main thread, continue
+                        main = thread.get("main", None)
+                        if main is None or main is False:
+                            continue
+
+                        id = thread.get("id", None)
+                        if id is not None and id == thread_id:
+                            ## if it's the main thread, mark it
+                            exception_main_thread = True
+                            break
```

Comment:
@evanh how do I know if this is ok and not heavy? usually, there are not a lot of exceptions or threads, but I don't know what's ok and what's not here.

Reply:
We monitor that performance pretty well, so I think mostly what you should do is let the on-call person know when this change is going out so they know to keep an extra eye out.

If we need too we can also add a metric to track the numbers later.